src/algorithms/ctmrg/ctmrg.jl

```diff
@@ -27,52 +29,51 @@ Different levels of output information are printed depending on `alg.verbosity`,
 suppresses all output, `1` only prints warnings, `2` gives information at the start and
 end, and `3` prints information every iteration.
 """
-function MPSKit.leading_boundary(state, alg::CTMRGAlgorithm)
-    return MPSKit.leading_boundary(CTMRGEnv(state, oneunit(spacetype(state))), state, alg)
+function MPSKit.leading_boundary(network, alg::CTMRGAlgorithm)
+    return MPSKit.leading_boundary(CTMRGEnv(network, oneunit(spacetype(network))), network, alg)
 end
 function MPSKit.leading_boundary(envinit, state, alg::CTMRGAlgorithm)
+    return MPSKit.leading_boundary(envinit, InfiniteSquareNetwork(state), alg)
+end
+function MPSKit.leading_boundary(envinit, network::InfiniteSquareNetwork, alg::CTMRGAlgorithm)
     CS = map(x -> tsvd(x)[2], envinit.corners)
     TS = map(x -> tsvd(x)[2], envinit.edges)
 
-    η = one(real(scalartype(state)))
+    η = one(real(scalartype(network)))
     env = deepcopy(envinit)
     log = ignore_derivatives(() -> MPSKit.IterLog("CTMRG"))
 
     return LoggingExtras.withlevel(; alg.verbosity) do
-        ctmrg_loginit!(log, η, state, envinit)
+        ctmrg_loginit!(log, η, network, envinit)
         local info
         for iter in 1:(alg.maxiter)
-            env, info = ctmrg_iteration(state, env, alg)  # Grow and renormalize in all 4 directions
+            env, info = ctmrg_iteration(network, env, alg)  # Grow and renormalize in all 4 directions
             η, CS, TS = calc_convergence(env, CS, TS)
 
             if η ≤ alg.tol && iter ≥ alg.miniter
-                ctmrg_logfinish!(log, iter, η, state, env)
+                ctmrg_logfinish!(log, iter, η, network, env)
                 break
             end
             if iter == alg.maxiter
-                ctmrg_logcancel!(log, iter, η, state, env)
+                ctmrg_logcancel!(log, iter, η, network, env)
             else
-                ctmrg_logiter!(log, iter, η, state, env)
+                ctmrg_logiter!(log, iter, η, network, env)
             end
         end
         return env, info
     end
 end
 
-# network-specific objective functions
-ctmrg_objective(state::InfinitePEPS, env::CTMRGEnv) = norm(state, env)
-ctmrg_objective(state::InfinitePartitionFunction, env::CTMRGEnv) = value(state, env)
```

Comment:
Since CTMRG now really runs on a concrete contractible network, the objective is just the value of that network. So I removed `ctmrg_objective` and just use `value` (and then implement the PEPS norm via `value` of the corresponding `InfiniteSquareNetwork`...).